doc/md clarify ent enum usage (#1886)
ent/ent: doc/md/schema-fields.md

}
```

## Enum Fields

> Use Ent API to generate Enum type and define possible values:
> ```go

Reviewer: My rephrase this or use this:

```suggestion
The `Enum` builder allows creating enum fields with a list of permitted values. 

```go
```

// Fields of the User.
func (User) Fields() []ent.Field {
	return []ent.Field{
		field.String("first_name"),
		field.String("last_name"),
		field.Enum("size").
			Values(
				"big",
				"small",
			),
	}
}
```

> When custom type that is convertible to the Go basic type:

Reviewer: ```suggestion
When a custom [`GoType`](#go-type) is being used, it is must be convertible to the basic `string` type or it needs to implement the [ValueScanner](https://pkg.go.dev/entgo.io/ent/schema/field#ValueScanner) interface. 

The [EnumValues](https://pkg.go.dev/entgo.io/ent/schema/field#EnumValues) interface is also required by the custom Go type to tell Ent what are the permitted values of the enum. 

The following example shows how to define an `Enum` field with a custom Go type that is convertible to `string`: 
```

> ```go

Reviewer: Merge the 2 code blocks to one. Easier to follow.

Author: I actually like the split, looks more like a recipe and easier for the reader to understand what is "in ent" and what Is part of my own type (outside ent).

// Fields of the User.
func (User) Fields() []ent.Field {
	return []ent.Field{
		field.String("first_name"),
		field.String("last_name"),
		// A convertible type to string.
		field.Enum("shape").
			GoType(property.Shape("")),
	}
}
```

Implement the [EnumValues](https://pkg.go.dev/entgo.io/ent/schema/field#EnumValues) interface.
```go
package property

type Shape string

const (
	Triangle Shape = "TRIANGLE"
	Circle   Shape = "CIRCLE"
)

// Values provides list valid values for Enum.

> func (Shape) Values() (roles []string) {
> 	for _, r := range []Shape{Triangle, Circle} {
> 		roles = append(roles, string(r))
> 	}
> 	return
> }

Reviewer 2: ```suggestion
func (Shape) Values() (kinds []string) {
	for _, r := range []Shape{Triangle, Circle} {
		kinds = append(kinds, string(r))
	}
	return
}
```

```

> When custom type is not convertible to the Go basic type:
> 

Reviewer: ```suggestion
The following example shows how to define an `Enum` field with a custom Go type that is not convertible to `string`, but it implements the [ValueScanner](https://pkg.go.dev/entgo.io/ent/schema/field#ValueScanner) interface: 

```

			GoType(property.Level(0)),
	}
}
```
Implement also the [ValueScanner](https://pkg.go.dev/entgo.io/ent/schema/field?tab=doc#ValueScanner) interface.

```go
package property

import "database/sql/driver"

type Level int

const (
	Unknown Level = iota
	Low
	High
)

func (p Level) String() string {
	switch p {
	case Low:
		return "LOW"
	case High:
		return "HIGH"
	default:
		return "UNKNOWN"
	}
}

// Values provides list valid values for Enum.
func (Level) Values() []string {
	return []string{Unknown.String(), Low.String(), High.String()}
}

// Value provides the DB a string from int.
func (p Level) Value() (driver.Value, error) {
	return p.String(), nil
}

// Scan tells our code how to read the enum into our type.
func (p *Level) Scan(val interface{}) error {
	var s string
	switch v := val.(type) {
	case nil:
		return nil
	case string:
		s = v
	case []uint8:
		s = string(v)
	}
	switch s {
	case "LOW":
		*p = Low
	case "HIGH":
		*p = High
	default:
		*p = Unknown
	}
	return nil
}
```

Combining it all together:
```go
// Fields of the User.
func (User) Fields() []ent.Field {
	return []ent.Field{
		field.String("first_name"),
		field.String("last_name"),
		field.Enum("size").

> 			Values(
> 				"big",
> 				"small",
> 			),

Reviewer: ```suggestion
			Values("big", "small"),
```

		// A convertible type to string.
		field.Enum("shape").
			GoType(property.Shape("")),

> 		// Add conversion to and from string

Reviewer: ```suggestion
		// Add conversion to and from string.
```

		field.Enum("level").
			GoType(property.Level(0)),
	}
}
```

After code generation usage is trivial:
```go 

> client.User.Create().
> 		SetFirstName("John").
> 		SetLastName("Dow").
> 		SetSize(user.SizeSmall).
> 		SetShape(property.Triangle).
> 		SetLevel(property.Low).
> 		SaveX(context.Background())
> 
> 	john := client.User.Query().FirstX(context.Background())
> 	fmt.Println(john) // User(id=1, first_name=John, last_name=Dow, size=small, shape=TRIANGLE, level=LOW)

Reviewer: ```suggestion
client.User.Create().
	SetFirstName("John").
	SetLastName("Dow").
	SetSize(user.SizeSmall).
	SetShape(property.Triangle).
	SetLevel(property.Low).
	SaveX(context.Background())
	
	john := client.User.Query().FirstX(context.Background())
	fmt.Println(john)
	// User(id=1, first_name=John, last_name=Dow, size=small, shape=TRIANGLE, level=LOW)
```

> 	john := client.User.Query().FirstX(context.Background())
> 	fmt.Println(john)
> 	// User(id=1, first_name=John, last_name=Dow, size=small, shape=TRIANGLE, level=LOW)

Reviewer: ```suggestion
john := client.User.Query().FirstX(context.Background())
fmt.Println(john)
// User(id=1, first_name=John, last_name=Dow, size=small, shape=TRIANGLE, level=LOW)
```
